src/AWSCredentials.jl

```diff
@@ -312,10 +393,7 @@ end
 
 function aws_get_role(role::AbstractString, ini::Inifile)
     source_profile, role_arn = aws_get_role_details(role, ini)
-
-    if source_profile == :notfound
-        error("Can't find AWS credentials!")
-    end
+    source_profile == :notfound && return nothing
```

Comment:
Perhaps use `===`? That's typically what should be used to compare `Symbol`s. (I realize this check is preexisting, but since you're touching the line anyway...)